source/extensions/filters/common/expr/context.cc

```diff
@@ -400,6 +400,18 @@ const UpstreamLookupValues& UpstreamLookupValues::get() {
           {UpstreamRequestAttemptCount,
            [](const UpstreamWrapper& wrapper) -> absl::optional<CelValue> {
              return CelValue::CreateUint64(wrapper.info_.attemptCount().value_or(0));
+          }},
+          {UpstreamNumEndpoints,
+           [](const UpstreamWrapper& wrapper) -> absl::optional<CelValue> {
+             if (wrapper.info_.upstreamClusterInfo().value_or(nullptr) !=
+                 nullptr) {
+               return CelValue::CreateUint64(wrapper.info_.upstreamClusterInfo()
+                                               .value()
+                                               .get()
+                                               ->endpointStats()
+                                               .membership_total_.value());
+             }
+             return CelValue::CreateUint64(0);
```

Comment:
Will `return {}` works here? If so, it is better than returning 0 to represent non-existence

Reply:
should. 
updated.